xwiki-commons-core/xwiki-commons-job/src/main/java/org/xwiki/job/JobManagerConfiguration.java

```diff
@@ -42,4 +44,36 @@
      * @since 7.2M2
      */
     int getJobStatusCacheSize();
+
+    /**
+     * @return the number of {@link GroupedJobInitializer} to keep in cache.
+     * @since 12.5RC1
+     */
+    @Unstable
+    default int getGroupedJobInitializerCacheSize()
+    {
+        return 10;
```

Comment:
Do you agree with this default value for the cache of groupedJobInitializer @tmortagne ?

Reply:
It's probably a bit low. I doubt GroupedJobInitializer use much memory so we could easily increase it a bit by default (like 100 maybe). Would need to check how many nodes are created after a few jobs are run in XWiki Standard. On extension side there is the root "extension" node and then 1 per namespace (so in practice 1 par wiki). It's potentially a lot more for the refactoring module since you have the root "refactoring" node and then one node for each element of each EntityReference which is manipulated.

Reply:
> It's probably a bit low. I doubt GroupedJobInitializer use much memory so we could easily increase it a bit by default (like 100 maybe).

OK let's do that: my first move was to put 1000 but it seemed a bit high back then :) Especially since I don't expect we'll have many GroupedJobInitializer, but maybe I'm wrong there.